### ISSUE TITLE
Date type Locale does not seem to work

### ISSUE THREAD

**Issue by dadoonet** [elastic/elasticsearch#25515]
Same as reported for Ingest: See #25513 
I came to this while working on a [discuss problem](https://discuss.elastic.co/t/indexation-formats-de-dates-en-francais/91573/1)

Here is a recreation script (tested on 5.4.2):

```
DELETE test
PUT test
{
  "mappings": {
    "doc": {
      "properties": {
        "date": {
          "type": "date",
          "format": "MMM y",
          "locale": "FRENCH"
        }
      }
    }
  }
}
PUT test/doc/1
{
  "date": "may 2017"
}
PUT test/doc/2
{
  "date": "mai 2017"
}
```

Gives:

```json
# DELETE test
{
  "acknowledged": true
}

# PUT test
{
  "acknowledged": true,
  "shards_acknowledged": true
}

# PUT test/doc/1
{
  "_index": "test",
  "_type": "doc",
  "_id": "1",
  "_version": 1,
  "result": "created",
  "_shards": {
    "total": 2,
    "successful": 1,
    "failed": 0
  },
  "created": true
}

# PUT test/doc/2
{
  "error": {
    "root_cause": [
      {
        "type": "mapper_parsing_exception",
        "reason": "failed to parse [date]"
      }
    ],
    "type": "mapper_parsing_exception",
    "reason": "failed to parse [date]",
    "caused_by": {
      "type": "illegal_argument_exception",
      "reason": "Invalid format: \"mai 2017\""
    }
  },
  "status": 400
}
```

**Comment by cbuescher** [comment]
@dadoonet when I look at the javadoc for `Locale` which we use to parse the locale parameter, the correct locale code for france seems to be "fr". When you uses this it rejects "may" and accepts "mai". I tried it your example with random garbage `"locale"` settings ant they seem to default to `Root` rather than throw an error. Maybe we should detect that and throw an exception, but unless I'm missing something the locale code here is simply wrong.

**Comment by dadoonet** [comment]
Great! `fr` is exactly what is needed here.
I believe it needs more documentation or as you said a way to reject unknown locale. I'm changing the label to `discuss` instead.

Thanks a lot @cbuescher

**Comment by dadoonet** [comment]
BTW this is weird to me that ingest does not seem to use the same way of defining locales...

This is failing for instance. I reported it in #25513.

```
POST _ingest/pipeline/_simulate
{
  "pipeline": {
    "processors": [
      {
        "date": {
          "field": "date",
          "locale": "fr",
          "formats": [
            "MMM y"
          ]
        }
      }
    ]
  },
  "docs": [
    {
      "_source": {
        "date": "May 2017"
      }
    },
    {
      "_source": {
        "date": "Mai 2017"
      }
    }
  ]
}
```

**Comment by cbuescher** [comment]
We discussed this in FixIt Friday and agreed that it would be preferable reject unknown locales here (and use the same resolution process in the Ingest Date Processor) but this will need some investigation of how to do it. Relabeling this accordingly.

**Comment by alexshadow007** [comment]
@dadoonet This was fixed by #26361

**Comment by jpountz** [comment]
Fixed via #26361. Thanks @alexshadow007 !
